rasa_core/training/structures.py

```diff
@@ -340,18 +338,29 @@ def _remove_unused_generated_cps(self, story_steps, overlapping_cps,
                            for cp_name in unused_cps
                            if cp_name.startswith(GENERATED_CHECKPOINT_PREFIX)}
 
+        k_to_remove = set()
         for k, step in story_steps.items():
             # changed all ends
             updated = step.create_copy(use_new_id=False)
             updated.start_checkpoints = self._checkpoint_difference(
                     updated.start_checkpoints, unused_overlapping_cps)
+            if step.start_checkpoints and not updated.start_checkpoints:
+                # remove story step if the generated checkpoints were unique
```

Comment:
I don't get this comment

Reply:
I meant  `remove story step if the generated checkpoints were the only ones`

Reply:
:+1: